fleet/agent_group_service.go

```diff
@@ -235,6 +237,37 @@ func (svc fleetService) RemoveAgentGroup(ctx context.Context, token, groupId str
 		svc.logger.Error("removing agents from group channel", zap.Error(errors.Wrap(ErrMaintainAgentGroupChannels, err)))
 	}
 
+	group, err := svc.agentGroupRepository.RetrieveByID(ctx, groupId, ownerID)
+	if err != nil {
+		return err
+	}
+
+	var lastRead uint64 = 1
+	for {
+		connectedAgents, err := svc.mfsdk.ThingsByChannel(token, group.MFChannelID, lastRead, limitThingsByChannel, false)
+		if err != nil {
+			return err
+		}
+
+		for _, agent := range connectedAgents.Things {
+			err := svc.mfsdk.DisconnectThing(agent.ID, group.MFChannelID, token)
+			if err != nil {
+				svc.logger.Error("error while disconnecting agent", zap.String("agent_name", agent.Name), zap.String("agent_id", agent.ID), zap.Error(err))
```

Comment:
we need to log the channel id as well or else we won't be able to investigate from the log message. let's make the message "error while disconnecting agent from channel"